### PR TITLE
Improves python check docs to use virtualenv and sort out PYTHONPATH when needed

### PR DESCRIPTION
**Pull request by scottopell**
### What does this PR do?
Improves the developer documentation when trying to run python checks locally

### Motivation
https://xkcd.com/1987/

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by hestonhoffman — APPROVED** [review]

**Review by chouetz — APPROVED** [review]
Only suggestions, feel free to take them into account

File: docs/dev/checks/README.md [hunk 1]
```diff
@@ -146,13 +155,50 @@ dev/dist
...
+
+1. `python3 -m pip install virtualenv`
+1. `virtualenv $GOPATH/src/github.com/DataDog/datadog-agent/venv`
+1. Activate the virtualenv (OS-dependent)
+1. `python3 -m pip install ~/dev/integrations-core/datadog_checks_base[deps]`
+1. `PYTHONPATH="$PWD/venv/lib/python3.10/site-packages:$PYTHONPATH" inv agent.run`
+
+
+#### Example for user install
+1. `python3 -m pip install --user ~/dev/integrations-core/datadog_checks_base[deps]`
```
Comment (gh123man) [inline docs/dev/checks/README.md:185]:
```suggestion
1. `python3 -m pip install --user ~/dev/integrations-core/'datadog_checks_base[deps]'`
```

I think it needs to be formatted like this. (did not work as-is on my machine in ZSH)

File: docs/dev/checks/README.md [hunk 2]
```diff
@@ -146,13 +155,50 @@ dev/dist
...
+
+In order for python to find this package, we must do two things:
+1. Install the `integrations-core` dependencies
+2. Set the `PYTHONPATH` env var correctly to find those dependencies
+
+`datadog-checks-base` must be installed as a pre-req, and you can install via a
+`--user` install or a `virtualenv` install, as long as you set your `PYTHONPATH`
+correctly.
+
+#### Example for virtualenv
```
[the remark this answers is not in the document]
Comment (scottopell) [reply docs/dev/checks/README.md:174]: Not a silly question, there's actually a note in `agent_dev_env.md` about this, and the answer is no. Only `virtualenv` is known to work, not `venv`.
I didn't test this claim though.

File: docs/dev/checks/README.md [hunk 3]
```diff
@@ -146,13 +155,50 @@ dev/dist
...
+(see also the notes in [../agent_dev_env.md](../agent_dev_env.md)):
+
+1. `python3 -m pip install virtualenv`
+1. `virtualenv $GOPATH/src/github.com/DataDog/datadog-agent/venv`
+1. Activate the virtualenv (OS-dependent)
+1. `python3 -m pip install ~/dev/integrations-core/datadog_checks_base[deps]`
```
[the remark this answers is not in the document]
Comment (scottopell) [reply docs/dev/checks/README.md:180]: yes it does, I can modify this to use a generic path marker, this is just my convention, don't think we recommend this anywhere in our docs.

**Review by sgnn7 — APPROVED** [review]
Left a couple of comments but LGTM

File: docs/dev/checks/README.md [hunk 4]
```diff
@@ -146,13 +155,50 @@ dev/dist
...
+
+`datadog-checks-base` must be installed as a pre-req, and you can install via a
+`--user` install or a `virtualenv` install, as long as you set your `PYTHONPATH`
+correctly.
+
+#### Example for virtualenv
+(see also the notes in [../agent_dev_env.md](../agent_dev_env.md)):
+
+1. `python3 -m pip install virtualenv`
+1. `virtualenv $GOPATH/src/github.com/DataDog/datadog-agent/venv`
```
[the remark this answers is not in the document]
Comment (scottopell) [reply docs/dev/checks/README.md:178]: Thanks for the note, let me include that

**Pull request merged**